matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt

```diff
@@ -953,7 +957,8 @@ internal class DefaultCryptoService @Inject constructor(
         val eventContent = event.content.toModel<RoomHistoryVisibilityContent>()
         eventContent?.historyVisibility?.let {
             cryptoStore.setShouldEncryptForInvitedMembers(roomId, it != RoomHistoryVisibility.JOINED)
-        }
+            cryptoStore.setShouldShareHistory(roomId, it.shouldShareHistory())
+        } ?: cryptoStore.setShouldShareHistory(roomId, false)
```

Comment:
I am not sure of the syntax. if setShouldEncryptForInvitedMembers signature change we could end up calling both test and elvis?